packages/typespec-test/test/batch_modular/generated/typespec-ts/src/api/pagingHelpers.ts

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  getPagedAsyncIterator,
+  PagedAsyncIterableIterator as CorePagedAsyncIterableIterator,
+  PagedResult,
+  PageSettings as CorePageSettings,
+} from "@azure/core-paging";
+import {
+  Client,
+  createRestError,
+  PathUncheckedResponse,
+} from "@azure-rest/core-client";
+import {
+  PageSettings,
+  PagedAsyncIterableIterator,
+} from "../models/pagingTypes.js";
+/**
+ * Helper type to extract the type of an array
+ */
+export type GetArrayType<T> = T extends Array<infer TData> ? TData : never;
+
+/**
+ * Helper type to infer the Type of the paged elements from the response type
+ * This type is generated based on the swagger information for x-ms-pageable
+ * specifically on the itemName property which indicates the property of the response
+ * where the page items are found. The default value is `value`.
+ * This type will allow us to provide strongly typed Iterator based on the response we get as second parameter
+ */
+export type PaginateReturn<TResult> = TResult extends
+  | {
+      body: { value?: infer TPage };
+    }
+  | {
+      body: { items?: infer TPage };
+    }
+  ? GetArrayType<TPage>
+  : Array<unknown>;
+
+export function buildPagedAsyncIterator<
+  TElement,
+  TResponse extends PathUncheckedResponse = PathUncheckedResponse
+>(
+  client: Client,
+  initialSendFunction: (...args: any[]) => PromiseLike<TResponse>,
+  deserializeFunction: (result: TResponse) => Promise<unknown>,
+  sendFunctionArgs: any[] = []
```

Comment:
Instead of passing both `initialSendFunction` and `sendFunctionArgs`, we could pass a `getInitialResponse` of type `() => PromiseLike<TResponse>` and instantiate it as `() => initialSendFunction(...sendFunctionArgs)`.

Also, `deserializeFunction` doesn't seem to be named meaningfully, we know the client will already do basic deserialization on the response before this function is called. Perhaps we could name it `processResponseBody`? I am open to alternatives as well.